core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider.go

```diff
@@ -395,14 +396,17 @@ func (p *logEventProvider) readLogs(ctx context.Context, latest int64, filters [
 			merr = errors.Join(merr, fmt.Errorf("failed to get logs for upkeep %s: %w", filter.upkeepID.String(), err))
 			continue
 		}
+
+		filteredLogs := p.filterLogsByContent(filter, logs)
```

Comment:
we'll likely need to extract this as a utility function somewhere as it will also be needed in log recoverer

Reply:
after [this](https://github.com/smartcontractkit/chainlink/pull/10344) is merged, we can put it in core/util